viritin/src/main/java/org/vaadin/viritin/form/AbstractForm.java

```diff
@@ -130,10 +130,18 @@ public void setEntity(T entity) {
     /**
      * @return true if bean has been changed since last setEntity call.
      */
-    public boolean hasChanges() {
+    protected boolean hasChanges() {
```

Comment:
Why make this protected? You can override it even if it is public.

Reply:
You are right, I did the change just before submitting the PR and did not really check the previous visibility. I will revert it to public.